buildvertical.targets

```diff
@@ -8,102 +8,92 @@
-          BeforeTargets="AnnotateProjectsWithConfiguration">
-    <ItemGroup>
-      <OmitProject Include="%(Project.Identity)" Condition="'%(Project.OmitReason)' != ''">
-        <OmitReason>%(Project.OmitReason)</OmitReason>
-      </OmitProject>
-      <Project Remove="@(OmitProject)" />
-    </ItemGroup>
+ <Target Name="AnnotateProjectsWithConfiguration" 
+         Returns="@(ProjectWithConfiguration)"> 
+    <PropertyGroup>
+      <_ProjectName>$(MSBuildProjectFullPath)</_ProjectName>
+    </PropertyGroup>
+    <Message Importance="Low" Text="$(MSBuildProjectFullPath), Configuration - $(_BestConfiguration)" />
+    <ItemGroup> 
+       <ProjectWithConfiguration Include="$(_ProjectName)" Exclude="@(OmitProject)"> 
+         <AdditionalProperties Condition="'$(_BestConfiguration)' != '' and '$(_BestConfiguration)' != 'None'">Configuration=$(_BestConfiguration);%(ProjectWithConfiguration.AdditionalProperties)</AdditionalProperties>
+         <UndefineProperties Condition="'$(_BestConfiguration)' == '' or '$(_BestConfiguration)' == 'None'">Configuration;%(ProjectWithConfiguration.UndefineProperties)</UndefineProperties>
+       </ProjectWithConfiguration>
+       <ProjectWithConfiguration Remove="@(OmitProject)" />
+     </ItemGroup>
   </Target>
 
   <!-- The initial shaking target for trimming down applicable projects for specified vertical -->
   <Target Name="DetermineProjectsConfiguration"
-          Inputs="%(Project.Identity)"
+          Inputs="$(MSBuildProjectFullPath)"
           Outputs="fake"
-          Returns="@(Project)"
-          DependsOnTargets="TrimUnconvertedProjects"
+          Returns="@(OmitProject)"
           BeforeTargets="AnnotateProjectsWithConfiguration">
     <PropertyGroup>
-      <_VerticalProject>%(Project.Identity)</_VerticalProject>
-      <_VerticalProjectName>%(Project.FullPath)</_VerticalProjectName>
-      <_OmitReason>%(Project.OmitReason)</_OmitReason>
+      <_VerticalProject>$(MSBuildProjectFullPath)</_VerticalProject>
     </PropertyGroup>
 
-    <!-- Clear the VerticalTargets item -->
-    <ItemGroup><VerticalTargets Remove="@(VerticalTargets)" /></ItemGroup>
+    <!-- Clear the _Configuration item -->
+    <ItemGroup><_Configuration Remove="@(_Configuration)" /></ItemGroup>
     <MSBuild Targets="FindBestConfiguration"
-             Projects="$(_VerticalProject)"
-             Condition="'$(_OmitReason)' == ''">
+             Projects="$(_VerticalProject)">
       <Output TaskParameter="TargetOutputs"
-              ItemName="VerticalTargets" />
+              ItemName="_Configuration" />
     </MSBuild>
 
     <PropertyGroup>
-      <_SelectedOSGroup>%(VerticalTargets.OSGroup)</_SelectedOSGroup>
-      <_SelectedTargetGroup>%(VerticalTargets.TargetGroup)</_SelectedTargetGroup>
-      <_KeepProject>%(VerticalTargets.NoBuildConfigurations)</_KeepProject>
+      <_BestConfiguration>%(_Configuration.Identity)</_BestConfiguration>
```

Comment:
This is making the assumption that we only have one configuration, which is OK for now but we will likely need address this later.